Source/Blazorise/Utils/Converters.cs

```diff
@@ -1,156 +1,227 @@
-
-        /// <summary>
-        /// Gets the min and max possible value based on the supplied value type
-        /// </summary>
-        /// <typeparam name="TValue">Value data type.</typeparam>
-        /// <returns>Returns the min and max value of supplied value type.</returns>
-        /// <exception cref="System.InvalidOperationException">Throws when value type is unknown.</exception>
-        public static (object, object) GetMinMaxValueOfType<TValue>()
-        {
-            var type = typeof( TValue );
-
-            switch ( type )
-            {
-                case Type byteType when byteType == typeof( byte ) || byteType == typeof( byte? ):
-                    return (byte.MinValue, byte.MaxValue);
-                case Type shortType when shortType == typeof( short ) || shortType == typeof( short? ):
-                    return (short.MinValue, short.MaxValue);
-                case Type intType when intType == typeof( int ) || intType == typeof( int? ):
-                    return (int.MinValue, int.MaxValue);
-                case Type longType when longType == typeof( long ) || longType == typeof( long? ):
-                    return (long.MinValue, long.MaxValue);
-                case Type floatType when floatType == typeof( float ) || floatType == typeof( float? ):
-                    return (float.MinValue, float.MaxValue);
-                case Type doubleType when doubleType == typeof( double ) || doubleType == typeof( double? ):
-                    return (double.MinValue, double.MaxValue);
-                case Type decimalType when decimalType == typeof( decimal ) || decimalType == typeof( decimal? ):
-                    return (decimal.MinValue, decimal.MaxValue);
-                case Type sbyteType when sbyteType == typeof( sbyte ) || sbyteType == typeof( sbyte? ):
-                    return (sbyte.MinValue, sbyte.MaxValue);
-                case Type ushortType when ushortType == typeof( ushort ) || ushortType == typeof( ushort? ):
-                    return (ushort.MinValue, ushort.MaxValue);
-                case Type uintType when uintType == typeof( uint ) || uintType == typeof( uint? ):
-                    return (uint.MinValue, uint.MaxValue);
-                case Type ulongType when ulongType == typeof( ulong ) || ulongType == typeof( ulong? ):
-                    return (ulong.MinValue, ulong.MaxValue);
-                default:
-                    throw new InvalidOperationException( $"Unsupported type {type}" );
-            }
-        }
-    }
-}
+﻿#region Using directives
+using System;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.Globalization;
+using System.Linq;
+#endregion
+
+namespace Blazorise.Utils
+{
+    public static class Converters
+    {
+        #region Constants
+
+        private static readonly Type[] SimpleTypes =

```

Comment:
What about nullable types?

Reply:
Also, byte, short, int, long, etc.

Reply:
`type.IsPrimitive` in the method **IsSimpleType**  takes care of byte, short, int, long etc.